src/service/video-manager-impl.js

```diff
@@ -284,43 +299,111 @@ class VideoEntry {
   }
 }
 
+/* @type {Promise<boolean>} */
+let supportsAutoplayCache_ = null;
+
 /**
- * Detects whether the platform even supports autoplay videos.
- * @param {!Window} win
- * @return {boolean}
+ * Detects whether autoplay is supported.
+ * Note that even if platfrom supports autoplay, users or browsers can disable
+ * autoplay to save data / battery. This function detects both platfrom support
+ * and when autoplay is disabled.
+ *
+ * Service dependencies are taken explicitly for testability.
+ *
+ * @private visible for testing.
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
+ * @param {!./platform-impl.Platform} platform
+ * @param {!./timer-impl.Timer} timer
+ * @param {!../mode.ModeDef} mode
+ * @return {!Promise<boolean>}
  */
-function platformSupportsAutoplay(win) {
-  // Do not support autoplay in amp-lite viewer
-  if (getMode(win).lite) {
-    return false;
+export function supportsAutoplay(ampdoc, platform, timer, mode) {
+
+  // Use cached result if available
+  if (supportsAutoplayCache_) {
+    return supportsAutoplayCache_;
   }
 
-  const platform = platformFor(win);
-  // non-mobile platforms always supported autoplay
-  if (!platform.isAndroid() && !platform.isIos()) {
-    return true;
+  // We do not support autoplay in amp-lite viewer regardless of platfrom
+  if (mode.lite) {
+    return supportsAutoplayCache_ = Promise.resolve(false);
   }
 
+  // Short-circuit for known unsupported versions on mobile
   const version = platform.getMajorVersion();
-  if (platform.isAndroid()) {
-    if (platform.isFirefox() && version >= 37) {
-      return true;
-    }
-    if (platform.isChrome() && version >= 53) {
-      return true;
-    }
+  if (platform.isChrome() && version < 53) {
```

Comment:
Would this break autoplaying on Desktop for both Chrome and Safari below? Why dropped the OS Check?